exporter/elasticsearchexporter/config.go

```diff
@@ -169,9 +169,13 @@ type RetrySettings struct {
 	// Enabled allows users to disable retry without having to comment out all settings.
 	Enabled bool `mapstructure:"enabled"`
 
-	// MaxRequests configures how often an HTTP request is retried before it is assumed to be failed.
+	// MaxRequests configures how often an HTTP request is attempted before it is assumed to be failed.
+	// Deprecated: use MaxRetries instead.
 	MaxRequests int `mapstructure:"max_requests"`
 
+	// MaxRetries configures how many times an HTTP request is retried.
+	MaxRetries int `mapstructure:"max_retries"`
```

Comment:
Is there a reason this is an `int` instead of `uint`?

Reply:
I think the libraries elasticsearchexporter uses all have their retries config as int, so it'll be straightforward for us to use int here as well.

Reply:
In that case I believe we should add validation then in `Validate()` to ensure it's a non-negative value.

Reply:
out of curiosity, is there a way to for the user to pass in 0 as an int value in the exporter config, other than using a `*int`? I believe it is indistinguishable from unset and it will be overwritten by a default.

Reply:
or to put it in a different way, what's the best practice if we want to accept 0 as a valid int configuration value?

Reply:
No, there's no way to distinguish if it's `int`.

Since you've called out how to properly disable retries in the README, I'd assume it's enough to not have to worry about users setting the `max_retries` to 0. If anyone ever comments about unexpected behavior, we can simply refer them to the `enabled` config option, and have them set it to false.

To me, it's fine if validate succeeds if `max_retries` is `0`, whether it was set by the user or not.